common/apitypes/txl2.go

```diff
@@ -142,9 +144,92 @@ func (tx TxL2) MarshalJSON() ([]byte, error) {
 
 // UnmarshalJSON is used to create a TxL2 from JSON data
 func (tx *TxL2) UnmarshalJSON(data []byte) error {
-	err := json.Unmarshal(data, tx)
+	type jsonToken struct {
```

Comment:
Maybe it will be a little bit better to init struct outside of the function? And it seems, that token struct and jsonToken structs are the same. Is it really needed to have the same struct?

Reply:
It is a intermediary struct. I think that it shouldn't be outside these functions because if not, everyone in the package could use it and it is worthless. Also, I've been checking the project and we do this in the same way as here.

Reply:
If you still think that we shouldn't do it in this way, I can propose create a function that contains the struct and return an empty object with type jsonFormat. What do you think??